src/flash.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __APP_FLASH_H__
+#define __APP_FLASH_H__
+
+#ifdef CONFIG_BOOTLOADER_MCUBOOT
+
+#include <zephyr/dfu/flash_img.h>
+#include <zephyr/dfu/mcuboot.h>
+#include <zephyr/types.h>
+
+int flash_img_prepare(struct flash_img_context *flash);
+
+#else /* CONFIG_BOOTLOADER_MCUBOOT */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+struct flash_img_context {
+	/* empty */
+};
+
+static inline int flash_img_prepare(struct flash_img_context *flash)
```

Comment:
To match the function below (or can use `flash` below).
```suggestion
static inline int flash_img_prepare(struct flash_img_context *ctx)
```

Reply:
Although `ctx` is probably better, because `flash` is close to `flush`

Reply:
Updated.